### ISSUE TITLE
Surface task scheduling results(success/failure) from SchedulerClient

### ISSUE THREAD

**Issue by cyuyang** [kagkarlsson/db-scheduler#326]
Currently SchedulerClient#schedule() function hides the failure if a task is created as duplicates. 
```
@Override
public <T> void schedule(TaskInstance<T> taskInstance, Instant executionTime) {
    boolean success = taskRepository.createIfNotExists(SchedulableInstance.of(taskInstance, executionTime));
    if (success) {
        notifyListeners(ClientEvent.EventType.SCHEDULE, taskInstance, executionTime);
    }
}
```

It would be nice if the failure (or the boolean result) could be handed back to the user and let the user decide what action to take upon duplicates. Sometimes, duplicates are caused by programming bugs that are hard to detect if the failure is not surfaced.

**Comment by kagkarlsson** [comment]
Agreed

**Comment by kagkarlsson** [comment]
TBH, I feel the safe thing to do here is to throw an exception. This should be quite unusual. And introduce another method `scheduleIfNotExists`/`trySchedule` that will return boolean indicating if it succeeded or not

**Comment by cyuyang** [comment]
> TBH, I feel the safe thing to do here is to throw an exception. This should be quite unusual. And introduce another method `scheduleIfNotExists`/`trySchedule` that will return boolean indicating if it succeeded or not

Won't it be a breaking change for people that rely on it being silently ignored? Alternatively, we can keep the existing behavior in the method, and add new ones as you suggested.

**Comment by kagkarlsson** [comment]
🎉 This issue has been resolved in `v14.0.3` ([Release Notes](https://github.com/kagkarlsson/db-scheduler/releases/tag/v14.0.3))
